src/lib.rs

```diff
@@ -178,17 +258,19 @@ impl PrometheusMetrics {
             &["endpoint", "method", "status"],
         )
         .unwrap();
-        registry
-            .register(Box::new(http_requests_duration_seconds.clone()))
-            .unwrap();
+        registry.register(Box::new(http_requests_duration_seconds.clone()))?;
```

Comment:
This is an interesting one. If this fails, should we just bomb out, so that we present the same "guarantees" as the `new` method?

Reply:
`register()` [returns error if this counter is already registered](https://docs.rs/prometheus/0.7.0/prometheus/struct.Registry.html#method.register) => it should not panic on `unwrap` in `new`, where new `Registry` is created

It should panic with custom `Registry` if I try to do this:
```rust
let metrics1 = PrometheusMetrics::new_with_registry(registry.clone(), 
                                                    "private_api", 
                                                    Some("/metrics")) .unwrap();
let metrics2 = PrometheusMetrics::new_with_registry(registry.clone(), 
                                                    "private_api", 
                                                    Some("/metrics")) 
                                                    .unwrap(); // << it should panic here because "private_api_http_requests_total" already exists
```